proto/distributed_cache.proto

```diff
@@ -29,9 +29,11 @@ message ReadResponse {
   bytes data = 1;
 }
 
+// Next Tag: 9
 message WriteRequest {
   reserved 1;
   bool finish_write = 3;
+  bool allow_already_exists = 8;
```

Comment:
allow_already_exists=True does the opposite of the name since it returns an error if something already exists. Perhaps change this to `check_already_exists` and add a comment describing what it does?